src/overlords/offense/roomPoisoner.ts

```diff
@@ -0,0 +1,159 @@
+import {log} from '../../console/log';
+import {Roles, Setups} from '../../creepSetups/setups';
+import {Pathing} from '../../movement/Pathing';
+import {DirectivePoisonRoom} from '../../directives/offense/poisonRoom';
+import {OverlordPriority} from '../../priorities/priorities_overlords';
+import {profile} from '../../profiler/decorator';
+import {Tasks} from '../../tasks/Tasks';
+import {Zerg} from '../../zerg/Zerg';
+import {MY_USERNAME} from '../../~settings';
+import {Overlord} from '../Overlord';
+import {OvermindConsole} from '../../console/Console';
+import {ColonyMemory} from '../../Colony';
+
+/**
+ * Spawn roomPoisoner - upgrqde controller to lvl2, wall in controller then sources.
+ */
+@profile
+export class RoomPoisonerOverlord extends Overlord {
+
+	roomPoisoners: Zerg[];
+	antiControllers: Zerg[]; //claim and counter reserve if required
+
+    controllerWallSites: ConstructionSite[] | undefined;
+    sourcesWallSites: ConstructionSite[] | undefined;
+
+	constructor(directive: DirectivePoisonRoom, priority = OverlordPriority.offense.roomPoisoner) {
+		super(directive, 'contaminate', priority);
+		this.roomPoisoners = this.zerg(Roles.roomPoisoner);
+		this.antiControllers = this.zerg(Roles.claim);
+
+		this.controllerWallSites = (this.room && this.room.controller) ? _.filter(this.room.constructionSites,
+                                                                    s => s.structureType == STRUCTURE_WALL &&
+                                                                    s.pos.isNearTo(this.room!.controller!.pos)) : undefined;
+    	this.sourcesWallSites = (this.room && this.room.controller) ? _.filter(this.room.constructionSites,
+                                                                    s => s.structureType == STRUCTURE_WALL &&
+                                                                    !s.pos.isNearTo(this.room!.controller!.pos)) : undefined;
+    }
+
+	refresh() {
+		super.refresh();
+		this.controllerWallSites = (this.room && this.room.controller) ? _.filter(this.room.constructionSites,
+            s => s.structureType == STRUCTURE_WALL &&
+            s.pos.isNearTo(this.room!.controller!.pos)) : undefined;
+        this.sourcesWallSites = (this.room && this.room.controller) ? _.filter(this.room.constructionSites,
+            s => s.structureType == STRUCTURE_WALL &&
+            !s.pos.isNearTo(this.room!.controller!.pos)) : undefined;
+	}
+
+	init() {
+		let isSafe = this.room && !this.room.dangerousPlayerHostiles.length;
+		let isNotReservedByEnemy = !(this.room && this.room.controller && this.room.controller.reservation && this.room.controller.reservation.ticksToEnd > 500);
```

Comment:
`isNotReservedByEnemy` is a little confusing since it adds double negatives. also, consider using `RoomIntel` reservation tracking methods here?

Reply:
made it isReservedByEnemy and used RoomIntel